keyring/backends/kwallet.py

```diff
@@ -46,6 +46,10 @@ def priority(cls):
             bus = dbus.SessionBus(mainloop=DBusGMainLoop())
         except dbus.DBusException as exc:
             raise RuntimeError(exc.get_dbus_message())
+        if not (bus.name_has_owner(cls.bus_name)
+                or cls.bus_name in bus.list_activatable_names()):
+            raise RuntimeError("The KWallet daemon is neither running nor "
+                               "activatable through D-Bus")
         try:
             bus.get_object(cls.bus_name, cls.object_path)
```

Comment:
I think we need to also remove this bit. As I understand it, the hang occurs when kwallet *is* an activatable name, but attempting to activate it fails. So the check you've added would pass (it would be in `list_activatable_names()`), and then it would try to do this all the same.

To avoid the hang, we have to not attempt to communicate with the service as part of this check. We *could* attempt this only if it's already running, but I think that's overcomplicating it.

Reply:
Removed.